src/reduction/mod.rs

```diff
@@ -1,3 +1,163 @@
-mod alpha;
+use std::collections::HashMap;
+use colors::{BitDepth, ColorType};
+use std::collections::hash_map::Entry::*;
+use png::PngData;
+use rgb::RGBA8;
+
+pub mod alpha;
 pub mod bit_depth;
 pub mod color;
+
+/// Fields to replace in PngData to apply the reduction
+pub struct ReducedPng {
+    pub color_type: ColorType,
+    pub raw_data: Vec<u8>,
+    /// replace if Some
+    pub palette: Option<Vec<RGBA8>>,
+    /// replace if Some
+    pub transparency_pixel: Option<Vec<u8>>,
+    /// replace if Some, delete if None
+    pub aux_headers: HashMap<[u8; 4], Option<Vec<u8>>>,
+}
+
+/// Attempt to reduce the number of colors in the palette
+/// Returns true if the palette was reduced, false otherwise
```

Comment:
The second line of this comment is outdated. Could you remove it please?

Reply:
Changed